### DISCUSSION TITLE
Bug? Annotated query incorrectly works for list[obj]

### DISCUSSION THREAD

**Discussion by nleononline** [fastapi/fastapi#10669] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [x] I commit to help with one of those options 👆

### Example Code

```python
from typing import Annotated

from fastapi import APIRouter, Depends, Query
from pydantic import BaseModel
from pydantic.json_schema import SkipJsonSchema


router = APIRouter(
    prefix="/chats",
    tags=["Chats"]
)


class ChatFilter(BaseModel):
    id: int | SkipJsonSchema[None] = None
    in_title: Annotated[list[str] | SkipJsonSchema[None], Query(...)] = None
    in_description: Annotated[list[str] | SkipJsonSchema[None], Query(...)] = None
    
    
@router.get("/")
async def get_chats(
    filters: Annotated[ChatFilter, Depends()] = None
):
    ...
```

### Description

OpenAPI JSON generated the in_title and in_description parameters in the body. If you replace BaseModel in ChatFilter with @dataclasses.dataclass, the parameters will be defined in the query.

### Operating System

macOS

### FastAPI Version

0.104.1

### Pydantic Version

2.4.2

### Python Version

3.11.4

**Comment by nleononline** [comment]
no answer?

**Reply by Kludex** [reply]
Please provide an MRE that I can run.

**Comment by JoshYuJump** [comment]
@n-leonovets Take a look at https://github.com/pydantic/pydantic/issues/7978

**Comment by YuriiMotov** [comment, marked as answer]
Since FastAPI 0.115.0 you can officially use models to define Query parameters.

You should update your code, change:
```python
async def get_chats(filters: Annotated[ChatFilter, Depends()] = None): ...
```
to
```python
async def get_chats(filters: Annotated[ChatFilter, Query()] = None): ...
```

So, the following code works well:

```python
from typing import Annotated

from fastapi import APIRouter, FastAPI, Query
from pydantic import BaseModel
from pydantic.json_schema import SkipJsonSchema

router = APIRouter(prefix="/chats", tags=["Chats"])

class ChatFilter(BaseModel):
    id: int | SkipJsonSchema[None] = None
    in_title: Annotated[list[str] | SkipJsonSchema[None], Query()] = None
    in_description: Annotated[list[str] | SkipJsonSchema[None], Query()] = None

@router.get("/")
async def get_chats(filters: Annotated[ChatFilter, Query()] = None): ...

app = FastAPI()
app.include_router(router)


```
